src/TTImp/TTImp.idr

```diff
@@ -496,26 +496,26 @@ findImplicits tm = []
 export
 implicitsAs : {auto c : Ref Ctxt Defs} ->
               Defs -> List Name -> RawImp -> Core RawImp
-implicitsAs defs ns tm = setAs (map Just (ns ++ map UN (findIBinds tm))) tm
+implicitsAs defs ns tm = setAs (map Just (ns ++ map UN (findIBinds tm))) [] tm
   where
-    setAs : List (Maybe Name) -> RawImp -> Core RawImp
-    setAs is (IApp loc f a)
-        = do f' <- setAs is f
+    setAs : List (Maybe Name) -> List (Maybe Name) -> RawImp -> Core RawImp
```

Comment:
Can we get a comment documenting the different roles of these two
`List (Maybe Name)` arguments?

Reply:
Sure, I'll add a comment in order to make it clearer to the reader!